statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,185 @@
+        self.vc = np.sqrt(np.diag(self.vc))
+
+    def _stat(self, SurveyDesign,c=None):
+        """                                                                                                                                              
+        Calculate a statistic with possible cluster deletion.                                                                                            
+                                                                                                                                                         
+        Parameters                                                                                                                                       
+        ----------                                                                                                                                       
+        c : int or None                                                                                                                                  
+            If an integer, return the statistic calculated with                                                                                          
+            cluster c deleted.  If c is None, return the statistic                                                                                       
+            calculated for the whole data set.                                                                                                           
+                                                                                                                                                         
+        Returns                                                                                                                                          
+        -------                                                                                                                                          
+        An array containing the statistic calculated on the columns                                                                                      
+        of the dataset.                                                                                                                                  
+        """
+
+        if c is None:
+            return np.dot(SurveyDesign.weights, self.data)
+
+        s = SurveyDesign.sfclust[c]
+        nh = SurveyDesign.ncs[s]
+        self.w = SurveyDesign.weights.copy()
+        self.w[SurveyDesign.strat == s] *= nh / float(nh - 1)
+        self.w[SurveyDesign.sclust == c] = 0
+        return np.dot(self.w, self.data)
+
+class SurveyPercentile(SurveyDesign):
+
+    def __init__(self, SurveyDesign, data, percentile):
+        self.data = data
+        self.cumsum_weights = np.cumsum(SurveyDesign.weights)
+        self.est = [[self._stat(perc, index) for index in range(self.data.shape[1])] for perc in percentile]
+
+    def _stat(self, percentile, col_index):
```

Comment:
`_stat` is called by `jack` in the parent class, so the signature has to be consistent across subclasses (therefore can't have percentile as an argument).  You could attach percentile in the subclass initializer, then refer to is as an attribute inside _stat.